src/defi_integrations/MorphoScripts.sol

```diff
@@ -0,0 +1,261 @@
+            (assetsRepaid, sharesRepaid) = IMorpho(morpho).repay(marketParams, assets, shares, onBehalf, data);
+            IERC20(marketParams.loanToken).forceApprove(morpho, 0);
+        } else {
+            revert InvalidInput();
+        }
+    }
+
+    /**
+     * @notice Supply collateral to a Morpho blue market on behalf of `onBehalf`
+     * @param morpho The address of the top level Morpho contract
+     * @param marketParams The market parameters of the individual morpho blue market
+     * @param assets The amount of assets to supply as collateral
+     * @param onBehalf The address of the account to supply collateral on behalf of
+     * @param data Arbitrary data to pass to the `onMorphoSupplyCollateral` callback. Pass empty data if not needed
+     */
+    function supplyCollateral(
+        address morpho,
+        MarketParams memory marketParams,
+        uint256 assets,
+        address onBehalf,
+        bytes calldata data
+    ) external {
+        IERC20(marketParams.collateralToken).forceApprove(morpho, assets);
+        IMorpho(morpho).supplyCollateral(marketParams, assets, onBehalf, data);
+    }
+
+    /**
+     * @notice Withdraw collateral from a Morpho blue market on behalf of `onBehalf` to `receiver`
+     * @param morpho The address of the top level Morpho contract
+     * @param marketParams The market parameters of the individual morpho blue market
+     * @param assets The amount of assets to withdraw as collateral
+     * @param onBehalf The address of the account to withdraw collateral on behalf of
+     * @param receiver The address of the account to receive the withdrawn collateral
+     */
+    function withdrawCollateral(
+        address morpho,
+        MarketParams memory marketParams,
+        uint256 assets,
+        address onBehalf,
+        address receiver
+    ) external {
+        IMorpho(morpho).withdrawCollateral(marketParams, assets, onBehalf, receiver);
+    }
+
+    /**
+     * @notice Repay assets and withdraw collateral from a Morpho blue market on behalf of `onBehalf` and send collateral to `receiver`
+     * @param morpho The address of the top level Morpho contract
+     * @param marketParams The market parameters of the individual morpho blue market
+     * @param repayAmount The amount of assets to repay
+     * @param repayShares The amount of shares to repay
+     * @param withdrawAmount The amount of assets to withdraw as collateral
+     * @param onBehalf The address of the account to repay and withdraw collateral on behalf of
+     * @param receiver The address of the account to receive the withdrawn collateral
+     */
+    function repayAndWithdrawCollateral(
+        address morpho,
+        MarketParams memory marketParams,
+        uint256 repayAmount,
+        uint256 repayShares,
+        uint256 withdrawAmount,
+        address onBehalf,
+        address receiver
+    ) external {
+        if (repayAmount > 0 || repayShares > 0) {
```

Comment:
This outer if-statement seems unnecessary

Reply:
Was mainly for saving actions on builder that client can simply just withdraw collateral by passing zero repay amount so the code could skip the repay clause without having to call separate script.

Reply:
The latest logics has been adjusted to: 
```
if (repayAmount > 0 || repayShares > 0) {
            if (repayAmount == type(uint256).max) {
                // Repay max
                IERC20(marketParams.loanToken).forceApprove(morpho, type(uint256).max);
                IMorpho(morpho).repay({});
                IERC20(marketParams.loanToken).forceApprove(morpho, 0);
            } else if (repayAmount > 0) {
                IERC20(marketParams.loanToken).forceApprove(morpho, repayAmount);
                IMorpho(morpho).repay({ });
            } else {
                IERC20(marketParams.loanToken).forceApprove(morpho, type(uint256).max);
                IMorpho(morpho).repay({});
                IERC20(marketParams.loanToken).forceApprove(morpho, 0);
            }
        }
```

So outer if became more like if user gives both 0, it's likely user just want to withdraw collateral,  so outer if skip the whole repay actions clause.